src/test/java/org/assertj/core/api/Assertions_assertThat_with_Iterator_Test.java

```diff
@@ -159,6 +159,7 @@ public void isNotSameAs_should_check_the_original_iterator_without_consuming_it(
     Assertions.fail("Expected assertionError, because assert notSame on same iterator.");
   }
 
+  /* TODO
   @Test
   public void iterator_can_be_asserted_twice_even_though_it_can_be_iterated_only_once() {
```

Comment:
on the contrary, withe incoming changes, iterator cannot be asserted twice, the test should show that

Reply:
Why should it not be asserted twice? All of my new assertions do not modify the iterator at all.